Update test object generation

raymondblanc opened the pull request:
-Added a test object generator that helps avoid redundant test object creation by providing methods that return custom and default objects based on input parameters.
-Added methods for creating test objects for : solr documents, log events, mail and slack alert profiles, alert rules, and triggered allerts

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/TestObjectGenerator.scala:
```diff
@@ -0,0 +1,63 @@
+                              message: String = "The service is down.",
+                              threadName: String = "thread3",
+                              throwable: String = "NullPointerException"
+
```
nit: extra space

raymondblanc commented:
Yes, I can do same for  logevent, alert, and triggers objects

> On Sep 10, 2018, at 11:27 AM, Anthony Foerster <notifications@github.com> wrote:
> 
> @afoerster requested changes on this pull request.
> 
> Looks good, are there any other objects you can replace now?
> 
> In alert-engine/src/test/scala/io/phdata/pulse/alertengine/TestObjectGenerator.scala <https://github.com/phdata/pulse/pull/55#discussion_r216387327>:
> 
> > +    * @param timestamp
> +    * @param level
> +    * @param message
> +    * @param threadName
> +    * @param throwable
> +    * @return solrDocument
> +    */
> +  def solrDocumentTestObject(
> +                              id: String = "123",
> +                              category: String = "test",
> +                              timestamp: String = "2018-04-06 10:15:00",
> +                              level: String = "FATAL",
> +                              message: String = "The service is down.",
> +                              threadName: String = "thread3",
> +                              throwable: String = "NullPointerException"
> +
> nit: extra space
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/phdata/pulse/pull/55#pullrequestreview-153856607>, or mute the thread <https://github.com/notifications/unsubscribe-auth/Am-OovWAihAXEsOAG-xQ638lGZrcdlhpks5uZpMMgaJpZM4WfItq>.
>

raymondblanc commented:
-Added a test object generator that helps avoid redundant test object creation by providing methods that return custom and default objects based on input parameters.
-Added methods for creating test objects for : solr documents, log events, mail and slack alert profiles, alert rules, and triggered alerts

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/AlertEngineImplTest.scala:
```diff
@@ -97,21 +75,24 @@ class AlertEngineImplTest
   test("get active alert") {
-    val alert = AlertRule("category: ERROR", 1, Some(0), List("tony@phdata.io"))
+    val a = TestObjectGenerator.alertRuleTestObject(query = "category: ERROR", retryInterval = 1, resultThreshold = Some(1), alertProfiles = List("tony@phdata.io"))
+    //val alertRule = AlertRule("category: ERROR", 1, Some(0), List("tony@phdata.io"))
```
Nit: remove comments

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/TestObjectGenerator.scala:
```diff
@@ -0,0 +1,161 @@
+    * @param properties
+    * @return LogEvent
+    */
+  def logEventTestObject(id: Option[String] = Some("id"),
```
`TestObject` is redundant here since the object is called `TestObjectGenerator`

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/TestObjectGenerator.scala:
```diff
@@ -0,0 +1,161 @@
+    * @param addresses
+    * @return MailAlertProfile
+    */
+  def mailAlertProfileTestObject(name: String = "mailprofile1",
```
Redundant `TestObject`

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/TestObjectGenerator.scala:
```diff
@@ -0,0 +1,161 @@
+    * @param totalNumFound
+    * @return TriggeredAlert
+    */
+  def triggeredAlertTestObject(rule: AlertRule = alertRuleTestObject(),
```
redundant `TestObject`

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/TestObjectGenerator.scala:
```diff
@@ -0,0 +1,161 @@
+    * @param alertProfiles
+    * @return AlertRule
+    */
+  def alertRuleTestObject(query: String = "id : testId",
```
redundant `TestObject`

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/AlertEngineConfigParserTest.scala:
```diff
@@ -46,9 +46,9 @@ class AlertEngineConfigParserTest extends FunSuite {
-            Some(List(MailAlertProfile("mailProfile1", List("tony@phdata.io")))),
-            Some(List(SlackAlertProfile("slackProfile1", "testurl.com")))
+            List(TestObjectGenerator.alertRuleTestObject(query = "query", retryInterval = 10, resultThreshold = Some(0), alertProfiles = List("mailProfile1", "slackProfile1"))),
+            Some(List(TestObjectGenerator.mailAlertProfileTestObject(name = "mailProfile1", addresses = List("tony@phdata.io")))),
```
Do me a favor and remove my email? 😄

raymondblanc replied:
Done! 😄.
Sorry about that.

raymondblanc commented:
-Refactored all redundant method names in the test object generator.
-Removed unnecessary comments.
-Updated alert profiles with a test email address

afoerster commented on alert-engine/src/test/scala/io/phdata/pulse/alertengine/AlertEngineImplTest.scala:
```diff
@@ -97,21 +75,22 @@ class AlertEngineImplTest
 
   test("get active alert") {
-    val alert = AlertRule("category: ERROR", 1, Some(0), List("tony@phdata.io"))
+    val a = TestObjectGenerator.alertRule(query = "category: ERROR", retryInterval = 1, resultThreshold = Some(1), alertProfiles = List("test@phdata.io"))
```
This should have a more descriptive name than 'a', change it back to 'alert'?

raymondblanc replied:
Refactored variable names and test profiles with more descriptive names
